From 2d422608609a39a6e5f90f404ae03c5c6f3864cc Mon Sep 17 00:00:00 2001
From: Keyon Jie <yang.jie@linux.intel.com>
Date: Fri, 22 Jun 2018 21:28:33 +0800
Subject: [PATCH 1238/1240] ASoC: core: don't override too much for BE
 dai_links

1. Don't override cpu_dai_name, we will reuse those defined in machine
driver.
2. We should keep dpcm_playback and dpcm_capture defined for BEs, not
both of them are supported by BEs sometimes.

Signed-off-by: Keyon Jie <yang.jie@linux.intel.com>
---
 sound/soc/soc-core.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/sound/soc/soc-core.c b/sound/soc/soc-core.c
index cd6608b..d34f422 100644
--- a/sound/soc/soc-core.c
+++ b/sound/soc/soc-core.c
@@ -2192,12 +2192,9 @@ static void soc_check_tplg_fes(struct snd_soc_card *card)
 
 			/* override platform */
 			dai_link->platform_name = platform->component.name;
-			dai_link->cpu_dai_name = platform->component.name;
 
 			/* convert non BE into BE */
 			dai_link->no_pcm = 1;
-			dai_link->dpcm_playback = 1;
-			dai_link->dpcm_capture = 1;
 
 			/* override any BE fixups */
 			dai_link->be_hw_params_fixup =
-- 
2.7.4

